huaweicloud/services/dli/resource_huaweicloud_dli_database_privilege.go

```diff
@@ -0,0 +1,244 @@
+// @API DLI GET /v1.0/{project_id}/authorization/privileges
+func ResourceDatabasePrivilege() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDatabasePrivilegeCreate,
+		ReadContext:   resourceDatabasePrivilegeRead,
+		UpdateContext: resourceDatabasePrivilegeUpdate,
+		DeleteContext: resourceDatabasePrivilegeDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDatabasePrivilegeImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The region in which to create the resource.`,
+			},
+			"object": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The authorization object definition.`,
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The name of the authorized (IAM) user.`,
+			},
+			"privileges": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: `The list of permissions granted to the database or data table.`,
+				DiffSuppressFunc: func(_, _, _ string, d *schema.ResourceData) bool {
+					if _, ok := d.GetOk("privileges"); !ok {
+						// If the value of the permission is nil, the Computed behavior will prevent changes.
+						// If the value of the permission is not nil and the length of the array is zero, need a
+						// DiffSuppress function to prevent changes.
+						return true
+					}
+					return false
+				},
+			},
+		},
+	}
+}
+
+func buildModifyDatabasePrivilegeBodyParams(d *schema.ResourceData, action string) map[string]interface{} {
+	// Default permission for database and data table.
+	privileges := []string{"SELECT"}
+	if d.Get("privileges").(*schema.Set).Len() > 0 {
+		privileges = utils.ExpandToStringListBySet(d.Get("privileges").(*schema.Set))
+	}
+
+	return map[string]interface{}{
+		"user_name": utils.ValueIngoreEmpty(d.Get("user_name")),
+		"action":    action,
+		"privileges": []map[string]interface{}{
+			{
+				"object":     d.Get("object"),
+				"privileges": privileges,
+			},
+		},
+	}
+}
+
+func modifyDatabasePrivilege(client *golangsdk.ServiceClient, d *schema.ResourceData, action string) error {
+	httpUrl := "v1.0/{project_id}/authorization"
+
+	modifyPath := client.Endpoint + httpUrl
+	modifyPath = strings.ReplaceAll(modifyPath, "{project_id}", client.ProjectID)
+
+	opts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildModifyDatabasePrivilegeBodyParams(d, action)),
+	}
+
+	requestResp, err := client.Request("PUT", modifyPath, &opts)
+	if err != nil {
+		return err
+	}
+
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return err
+	}
+	if !utils.PathSearch("is_success", respBody, "").(bool) {
```

Comment:
type convert error

Reply:
yeah, the default value should be true, not empty string.